### DISCUSSION TITLE
How to get GraphQL query time taken?

### DISCUSSION THREAD

**Discussion by GNevsky** [webonyx/graphql-php#1451] [Q&A]
I’m trying to get full time taken by GraphQL query on server side, like: 
```
...
$server = new StandardServer($config);
$startTime = microtime(true);
$server->handleRequest($parsedRequestBody);
$endTime = microtime(true);
$logger->write($endTime-$startTime);
...
```
But I’m getting only a small piece of time of full query which may consists of several queries/resolvers. 
Is there a way to get full time?

**Comment by le0daniel** [comment]
In my opinion the easiest way to do this is by decorating all your resolve functions. In my own toolkit (https://github.com/le0daniel/graphql-tools) that I wrote for my GraphQL applications i wrap each resolver by a class called ProxyResolver (which is invokable), which allows me to get detailed telemetry into every resolver and how long it takes in NanoSeconds... 

A similar approach could work for you if you need more detailed insights into what takes how long...
